src/cylinder.h

```diff
@@ -10,225 +10,44 @@
-	 * \returns ReturnValue holds the return value
-	 */
-	virtual ReturnValue queryRemove(const Thing& thing, uint32_t count, uint32_t flags, Creature* = nullptr) const = 0;
-
-	/**
-	 * Query the destination cylinder
-	 * \param index points to the destination index (inventory slot/container
-	 * position), -1 is a internal value and means add to a empty position, with
-	 * no destItem this method can change the index to point to the new cylinder
-	 * index \param destItem is the destination object \param flags optional
-	 * flags to modify the default behaviour this method can modify the flags
-	 * \returns Cylinder returns the destination cylinder
-	 */
-	virtual Cylinder* queryDestination(int32_t& index, const Thing& thing, Item** destItem, uint32_t& flags) = 0;
-
-	/**
-	 * Add the object to the cylinder
-	 * \param thing is the object to add
-	 */
-	virtual void addThing(Thing* thing) = 0;
-
-	/**
-	 * Add the object to the cylinder
-	 * \param index points to the destination index (inventory slot/container
-	 * position) \param thing is the object to add
-	 */
-	virtual void addThing(int32_t index, Thing* thing) = 0;
-
-	/**
-	 * Update the item count or type for an object
-	 * \param thing is the object to update
-	 * \param itemId is the new item id
-	 * \param count is the new count value
-	 */
-	virtual void updateThing(Thing* thing, uint16_t itemId, uint32_t count) = 0;
-
-	/**
-	 * Replace an object with a new
-	 * \param index is the position to change (inventory slot/container
-	 * position) \param thing is the object to update
-	 */
-	virtual void replaceThing(uint32_t index, Thing* thing) = 0;
-
-	/**
-	 * Remove an object
-	 * \param thing is the object to delete
-	 * \param count is the new count value
-	 */
-	virtual void removeThing(Thing* thing, uint32_t count) = 0;
-
-	/**
-	 * Is sent after an operation (move/add) to update internal values
-	 * \param thing is the object that has been added
-	 * \param index is the objects new index value
-	 * \param link holds the relation the object has to the cylinder
-	 */
-	virtual void postAddNotification(Thing* thing, const Cylinder* oldParent, int32_t index,
-	                                 cylinderlink_t link = LINK_OWNER) = 0;
-
-	/**
-	 * Is sent after an operation (move/remove) to update internal values
-	 * \param thing is the object that has been removed
-	 * \param index is the previous index of the removed object
-	 * \param link holds the relation the object has to the cylinder
-	 */
-	virtual void postRemoveNotification(Thing* thing, const Cylinder* newParent, int32_t index,
-	                                    cylinderlink_t link = LINK_OWNER) = 0;
-
-	/**
-	 * Gets the index of an object
-	 * \param thing the object to get the index value from
-	 * \returns the index of the object, returns -1 if not found
-	 */
-	virtual int32_t getThingIndex(const Thing* thing) const;
+	static std::shared_ptr<VirtualCylinder> virtualCylinder;
```

Comment:
Could this be `const`
```suggestion
	static const std::shared_ptr<VirtualCylinder> virtualCylinder;
```